e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,318 @@
+	if err := nt.Validate("hello-world", "dir", &appsv1.Deployment{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Remove the deployment")
+	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("check that the deployment was removed and replicaset remains")
+	if err := nt.ValidateNotFound("hello-world", "dir", &appsv1.Deployment{}); err != nil {
+		nt.T.Fatal(err)
+	}
+	_, err := nomostest.Retry(60*time.Second, func() error {
+		err := nt.Validate("hello-world", "dir", &appsv1.ReplicaSet{}, nomostest.HasLabel("app", "hello-world"))
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestRolebindingsUpdated(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2)
+
+	nt.RootRepos[configsync.RootSyncName].Copy("../../examples/acme/namespaces/eng/backend/namespace.yaml", "acme/namespaces/eng/backend/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy("../../examples/acme/namespaces/eng/backend/bob-rolebinding.yaml", "acme/namespaces/eng/backend/br.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add bob rolebinding")
+	nt.WaitForRepoSyncs()
+	if err := nt.Validate("bob-rolebinding", "backend", &rbacv1.RoleBinding{}, nomostest.RoleBindingsHasName("acme-admin")); err != nil {
+		nt.T.Fatal("bob-rolebinding not found")
+	}
+
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/robert-rolebinding.yaml", yamlDir), "acme/namespaces/eng/backend/br.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Replace bob with robert rolebinding")
+	nt.WaitForRepoSyncs()
+
+	if err := nt.ValidateNotFound("bob-rolebinding", "backend", &rbacv1.RoleBinding{}); err != nil {
+		nt.T.Fatal("bob-rolebinding is not deleted")
+	}
+
+	if err := nt.Validate("robert-rolebinding", "backend", &rbacv1.RoleBinding{}); err != nil {
+		nt.T.Fatal("robert-rolebinding not found")
+	}
+}
+
+func manageNamespace(nt *nomostest.NT, namespace string) {
+	nt.T.Log("Add an unmanaged resource into the namespace as a control")
+	nt.T.Log("We should never modify this resource")
+	_, err := nt.Kubectl("apply", "-f", fmt.Sprintf("%s/reserved_namespaces/unmanaged-service.%s.yaml", yamlDir, namespace))
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Add resource to manage")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/service.yaml", yamlDir), fmt.Sprintf("acme/namespaces/%s/service.yaml", namespace))
+	nt.T.Log("Start managing the namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Start managing the namespace")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Wait until managed service appears on the cluster")
```

Comment:
`Validate managed service appears on the cluster` (since the retry is removed)

ditto for the log statements below

Reply:
updated logs for clarity